make_prg/subcommands/from_msa.py

```diff
@@ -52,95 +75,94 @@ def register_parser(subparsers):
         ),
     )
     subparser_msa.add_argument(
-        "-o",
-        "--outdir",
-        dest="output_dir",
-        action="store",
-        default=".",
-        help="Output directory. Default: %(default)s",
-    )
-    subparser_msa.add_argument(
-        "-n",
-        "--prg_name",
-        dest="prg_name",
-        action="store",
-        help="Prg file name. Default: MSA file name",
-    )
-    subparser_msa.add_argument(
-        "-S",
-        "--seqid",
-        help="Sequence identifier to use for the output sequence/PRG. Default is the file name",
-    )
-    subparser_msa.add_argument(
-        "--no_overwrite",
-        dest="no_overwrite",
+        "--output_graphs",
+        dest="output_graphs",
         action="store_true",
-        help="Do not replace an existing prg file",
+        default=False,
+        help="Outputs the recursive tree graphical representation (for development use only)",
     )
-    subparser_msa.add_argument(
-        "-O",
-        "--output-type",
-        help="p: PRG, b: Binary, g: GFA, a: All. Combinations are allowed i.e., gb: GFA and Binary. Default: %(default)s",
-        default="a",
-        type=OutputType,
-    )
-    subparser_msa.add_argument("--log", help="Path to write log to. Default is stderr")
+
     subparser_msa.set_defaults(func=run)
 
     return subparser_msa
 
 
-def run(options):
-    MSA_file = Path(options.MSA).resolve()
-    if not MSA_file.exists():
-        raise ValueError(f"File not found: {options.MSA}")
-    output_dir = Path(options.output_dir).resolve()
-    output_dir.mkdir(parents=True, exist_ok=True)
+def get_all_input_files(input_path: str) -> List[Path]:
+    input_path = Path(input_path)
+    if not input_path.exists():
+        raise FileNotFoundError(f"{input_path} does not exist")
 
-    if options.prg_name is None:
-        options.prg_name = MSA_file.stem
-    ofile_prefix = output_dir / options.prg_name
+    if input_path.is_file():
+        all_files = [input_path]
+    else:
+        all_files = [
+            Path(path).absolute() for path in input_path.iterdir() if path.is_file()
+        ]
+    return all_files
 
-    # Set up file logging
-    formatter = logging.Formatter(
-        fmt="%(levelname)s %(asctime)s %(message)s", datefmt="%d/%m/%Y %I:%M:%S"
-    )
-    handler = (
-        logging.FileHandler(options.log) if options.log else logging.StreamHandler()
-    )
-    handler.setFormatter(formatter)
-    logging.getLogger().addHandler(handler)
 
-    logging.info(
-        "Input parameters max_nesting: %d, min_match_length: %d",
-        options.max_nesting,
-        options.min_match_length,
-    )
+def process_MSA(msa_filepath: Path):
+    global options
+    logger.info(f"Generating PRG for {msa_filepath}...")
+    msa_name = msa_filepath.name
+    locus_name = msa_filepath.with_suffix("").name
```

Comment:
This implies we don't accept/expect compressed files?

Reply:
You mean input like `gene.fasta.gz`? We don't accept this now, but we can add this feature, won't be hard. Should we do it for this PR, or do we leave a TODO for a next PR?

Reply:
Urgh, just saw biopython doesn't accept gzipped fasta. How spread out is our MSA file-reading functionality? Would it only require changing one/two lines? Or is it spread out over multiple modules?

Reply:
All MSA-file reading functionality are concentrated on [make_prg.io_utils.load_alignment_file](https://github.com/leoisl/make_prg/blob/698827c1c6b7200b657587b6af8610ceb71e762f/make_prg/utils/io_utils.py#L18-L28), which already supports reading `.gz` files. So, we just need to fix other code that was expecting just `.fa` files, which should not be hard, will be done for this PR. Thanks, I totally overlooked this!

Reply:
Sorry for the delay, was on a break. Done in https://github.com/iqbal-lab-org/make_prg/pull/35/commits/4e5492999223c7700502c22b0fb3060894b5953c , please check. Omitted here are two integration tests to ensure `make_prg from_msa` is reading compressed files correctly, will be described in a future PR.